### PR TITLE
MultiGet should not fail entirely if alias resolves to many indices

### LINKED ISSUE #20845: MultiGet fails entirely when using alias that points to multiple indices
When one of the items of a multi get request refers to an alias that points to multiple indices, only that specific item should fail (as we don't know where to get the document from). Instead, the whole request fails at the moment.

### PR DESCRIPTION
**Pull request by tlrx**
MultiGet should not fail entirely when one of the items of a multi get request refers to an alias that points to multiple indices.

closes #20845

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]

**Comment by nik9000** [comment]
Thanks for fixing the line lengths while you were there!

**Comment by nik9000** [comment]
Should this also go to `v5.1.0`?

**Comment by tlrx** [comment]
> Should this also go to v5.1.0?

Yes it should, I'll do it once CI is OK.

**Comment by tlrx** [comment]
retest this please

File: core/src/main/java/org/elasticsearch/action/get/TransportMultiGetAction.java [hunk 1]
```diff
@@ -47,45 +47,56 @@
...
+            try {
+                item.routing(clusterState.metaData().resolveIndexRouting(item.parent(), item.routing(), item.index()));
+                concreteSingleIndex = indexNameExpressionResolver.concreteSingleIndex(clusterState, item).getName();
+
+                if ((item.routing() == null) && (clusterState.getMetaData().routingRequired(concreteSingleIndex, item.type()))) {
+                    String message = "routing is required for [" + concreteSingleIndex + "]/[" + item.type() + "]/[" + item.id() + "]";
+                    responses.set(i, newItemFailure(concreteSingleIndex, item.type(), item.id(), new IllegalArgumentException(message)));
+                    continue;
+                }
+            } catch (IllegalArgumentException e) {
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/get/TransportMultiGetAction.java]: I wonder if this should be `Exception`. see also #20659

Comment (tlrx) [inline core/src/main/java/org/elasticsearch/action/get/TransportMultiGetAction.java]: Sure - I did not see #20659 before creating this pull request :(

**Comment by tlrx** [comment]
Thanks @nik9000 and @javanna

**Comment by javanna** [comment]
I think this should have been backported to 2.4 too? and to 5.0 ?

**Comment by tlrx** [comment]
@javanna Sorry, it looks like I forgot to backport it.

**Pull request merged**